docs/content/docs/contributor-docs/code-generator-config.md

```diff
@@ -694,19 +706,83 @@ We set this `requeue_on_success_seconds` value to `60` here because the values o
+
+NOTE: The reason for this is because the EC2 controller only deals with single Instance resources, not batches of instances
+
+```yaml
+resources:
+  Instance:
+    fields:
+      MaxCount:
+        is_required: false
+      MinCount:
+        is_required: false
+```
 
 ### Controlling a field's Go type
 
-**TODO**
+Use the `resources[$resource].fields[$field].type` configuration option to override a field's Go type. You will typically use this configuration option for custom fields that are not inferred by `ack-generate` by looking at the AWS API model definition.
+
+An example of this is the Policies field for a Role custom resource definition in the IAM controller. The IAM controller uses some custom hook code to allow a Kubernetes user to specify one or more Policy ARNs for a Role simply by specifying `Spec.Policies`. To define this custom field as a list of string pointers, the IAM controller's [`generator.yaml`](https://github.com/aws-controllers-k8s/iam-controller/blob/b6a62ca48c1aea7ab78e62fb3ad6845335c1f1c2/generator.yaml#L164-L168) file uses the following:
+
+```yaml
+resources:
+  Role:
+    fields:
+      # In order to support attaching zero or more policies to a Role, we use
+      # custom update code path code that uses the Attach/DetachGroupPolicy API
+      # calls to manage the set of PolicyARNs attached to this Role.
+      Policies:
+        type: "[]*string"
+```
 
 ### Controlling how a field's values are compared
 
-**TODO**
+Use the `resources[$resource].fields[$field].compare` configuration option to control how the value of a field is compared between two resources. This configuration option has two boolean subfields, `is_ignored` and `nil_equals_zero_value`.
+
+Use the `is_ignored` subfield to instruct the code generator to exclude this particular field from automatic value comparisons when building the `Delta` struct that compares two resources.
+
+Typically, you will want to mark a field as ignored for comparison operations because the Go type of the field does not natively support deterministic equality operations. For example, a slice of `Tag` structs where the code generator does not know how to sort the slice means that the default `reflect.DeepEqual` call will produce non-deterministic results. These types of fields you will want to mark with `compare.is_ignored: true` and include a custom comparison function using the `delta_pre_compare` hook, as this example from the IAM controller's [`generator.yaml`](https://github.com/aws-controllers-k8s/iam-controller/blob/b6a62ca48c1aea7ab78e62fb3ad6845335c1f1c2/generator.yaml#L172-L174) does for the Role resource:
+
+```yaml
+  Role:
+    hooks:
+      delta_pre_compare:
+        code: compareTags(delta, a, b)
+    fields:
+      Tags:
+        compare:
+          is_ignored: true
+```
+
+Use the `nil_equals_zero_value` subfield to instruct the code generator to treat a nil value the same as the zero value of the underlying Go type.
```

Comment:
I'm not sure if this is implemented yet

Reply:
Yeah, it's not used in any of the generator.yaml files... I suppose I can remove it from these docs.